torch/utils/checkpoint.py

```diff
@@ -405,16 +422,19 @@ def checkpoint(
         )
 
     if use_reentrant:
-        if context_fn is not noop_context_fn:
+        if context_fn is not noop_context_fn or debug is not False:
```

Comment:
Should you check determinism mode here?

Reply:
we should just update the docs to say "default" means nothing is checked for use_reentrant=True
I don't know if I want to force users to have to explicitly specify determinism_check="none" when using use_reentrant=True

Reply:
Ho that's because that's the default value. I would agree that we don't want to break all the current users yes :D 
I would say error on non none or default then yes.